src/OVAL/probes/unix/linux/rpmverifypackage.c

```diff
@@ -275,23 +309,48 @@ static int rpmverify_collect(probe_ctx *ctx,
 	return (ret);
 }
 
-void probe_preload ()
+void probe_offline_mode ()
 {
-	rpmLibsPreload();
+	probe_setoption(PROBEOPT_OFFLINE_MODE_SUPPORTED, PROBE_OFFLINE_OWN);
 }
 
 void *probe_init (void)
 {
-	probe_setoption(PROBEOPT_OFFLINE_MODE_SUPPORTED, PROBE_OFFLINE_CHROOT);
+	const char* root = getenv("OSCAP_PROBE_ROOT");
+	if ((root!= NULL) && (strlen(root) == 0)) {
+		root = NULL;
+	}
+	probe_chroot_init(&g_rpm.chr, root);
 
 	rpmlogSetCallback(rpmErrorCb, NULL);
-	if (rpmReadConfigFiles ((const char *)NULL, (const char *)NULL) != 0) {
+
+
+
+	if (CHROOT_IS_SET()) {
+		rpmLibsPreload();
+		if (CHROOT_ENTER() < 0) {
+			return (NULL);
+		}
+	}
+
+	if (rpmReadConfigFiles (NULL, (const char *)NULL) != 0) {
 		dI("rpmReadConfigFiles failed: %u, %s.", errno, strerror (errno));
 		return (NULL);
 	}
 
 	g_rpm.rpmts = rpmtsCreate();
 
+	if (CHROOT_IS_SET()) {
```

Comment:
I wonder why this block is not placed in probe_fini(). Could you explain the reason, please?